.github/workflows/pre-merge-integration-test.yml

```diff
@@ -29,7 +29,7 @@ jobs:
         uses: aws-actions/setup-sam@v2
 
       - name: Assume temporary AWS role
-        uses: aws-actions/configure-aws-credentials@v1-node16
+        uses: aws-actions/configure-aws-credentials@v1-node18
```

Comment:
We can't connect to AWS from anything other than the main branch, so doing this action in a PR workflow will not work.